Samples/AdventureDotNetCli/AdventureGrains/PlayerGrain.cs

```diff
@@ -12,17 +13,43 @@ public class PlayerGrain : Orleans.Grain, IPlayerGrain
     {
         IRoomGrain roomGrain; // Current room
         List<Thing> things = new List<Thing>(); // Things that the player is carrying
+        private ObserverSubscriptionManager<IMessage> _subsManager;
+
 
         bool killed = false;
 
         PlayerInfo myInfo;
 
+
         public override Task OnActivateAsync()
         {
             this.myInfo = new PlayerInfo { Key = this.GetPrimaryKey(), Name = "nobody" };
+
+
+            // Hook up stream where we push messages to Client
+            _subsManager = new ObserverSubscriptionManager<IMessage>();
+
             return base.OnActivateAsync();
         }
 
+        // Clients call this to subscribe.
+        public async Task Subscribe(IMessage observer)
+        {
+             _subsManager.Subscribe(observer);
+        }
+
+        //Also clients use this to unsubscribe themselves to no longer receive the messages.
+        public async Task UnSubscribe(IMessage observer)
```

Comment:
No need for ```async``` keyword, and the method should return ```TaskDone.Done``` to fultil the async API here.